### DISCUSSION TITLE
Hang oracledb.connect_async when limit for database session is reached

### DISCUSSION THREAD

**Discussion by jandvorak-sol60279** [oracle/python-oracledb#413] [Q&A]
i was rewriting some code used in fastapi for gathering data from sync to async version, but occasionally, when database got sessions depleted (i guess we have some leakage in java application using oracle datasource), the connect_async will hang with forever, with no raising exceptions at all.

using oracle onprem 19.22 and latest oracledb 2.4.1

while running regular oracledb.connect will raise

```
DatabaseError: DPY-4011: the database or network closed the connection
Help: https://python-oracledb.readthedocs.io/en/latest/user_guide/troubleshooting.html#dpy-4011
```
and then

```OperationalError: DPY-6005: cannot connect to database (CONNECTION_ID=IVTDeA2S6SCxrPt5363vJA==).
DPY-4011: the database or network closed the connection
Help: https://python-oracledb.readthedocs.io/en/latest/user_guide/troubleshooting.html#dpy-4011```

**Comment by jandvorak-sol60279** [comment]
workaround is to wrap connect_async with `asyncio.wait_for()` to explicitly set timeout and end it with `TimeoutError` but in my implementation, it add overhead of timeout value (no big deal atm)

**Comment by cjbj** [comment]
If you have a nice little, runnable test case, can you open an issue?

**Comment by anthony-tuininga** [comment]
This may be related to #392 which I am working on resolving. Once I have that resolved you can try your scenario and see if that helps.

**Reply by anthony-tuininga** [reply]
I have corrected that issue. Please check to see if that corrects yours as well. Thank you!

**Reply by cjbj** [reply]
@jandvorak-sol60279 did the fix (in python-oracledb 2.5) resolve your issue?

**Comment by jandvorak-sol60279** [comment]
i wasnt able to reproduce the issue and it stops occuring, so i suppose, it was resolved.
